examples/platforms/efr32/radio.c

```diff
@@ -771,6 +939,14 @@ static void RAILCb_Generic(RAIL_Handle_t aRailHandle, RAIL_Events_t aEvents)
         status = RAIL_Calibrate(aRailHandle, NULL, RAIL_CAL_ALL_PENDING);
         assert(status == RAIL_STATUS_NO_ERROR);
     }
+
+    if (aEvents & RAIL_EVENT_RSSI_AVERAGE_DONE)
+    {
+        sSampleRssiDone = false;
```

Comment:
Is it possible for `sSampleRssiDone` to get stuck as `true`? For example, is there anything that can abort this process?

Reply:
I just checked the `RAIL` API doc, it doesn't say whether the chip will generate an interrupt when the RSSI averaging is interrupted. So it has the possibility to get stuck. To prevent the code from getting stuck, I will add a timeout mechanism.